content/blog/copilot-lessons/index.md

```diff
@@ -0,0 +1,129 @@
+---
+title: "Engineering Lessons from Building Pulumi Copilot"
+date: 2024-12-12T13:56:13-05:00
+draft: false
+meta_desc: Learn key engineering lessons from building Pulumi Copilot, including how to minimize LLM workload, validate outputs, and deal with hallucination.
+meta_image: meta.png
+authors:
+    - artur-laksberg
+    - adam-gordon-bell
+tags:
+    - Copilot
+    - ai
+    - iac
+social:
+    twitter: 🚀 Building Pulumi Copilot taught us key AI engineering insights – Minimize LLM workload, validate outputs rigorously & use "skills" for modular tasks. Plus, sometimes when AI hallucinates features, it reveals exactly what users want! 
+    linkedin: 🚀 Building Pulumi Copilot taught us key AI engineering insights – Minimize LLM workload, validate outputs rigorously & use "skills" for modular tasks. Plus, sometimes when AI hallucinates features (like a "--force" flag), it reveals exactly what users want! 
+---
+**Artur Laksberg is on the Copilot team here at Pulumi. Copilot is an AI assistant for cloud infrastructure. Today he shares key lessons from bringing Copilot to production—including an unexpected discovery about AI hallucinations.**
```

Comment:
I'd like to re-do the introduction... It starts with third person narration ("Artur Laksberg is...") then switches to first person singular notation ("I braced for the worst...") and then continues with first person plural ("We learned a key lesson...")

This feels a bit jarring. I suggest keeping first person plural throughout. We both work in Pulumi, we tell the story.

Reply:
The idea is to sound like you explaining some tips and tricks for building on LLMs to another engineer in-person. 

In doing so, you might refer to team efforts as something "we" did, but I don't think you'd refer to yourself in plural: "we were reading our email and we were eating a sandwich"  :)

Reply:
LOL